src/openssl/lib_crypto.cr

```diff
@@ -1,11 +1,23 @@
 {% begin %}
   lib LibCrypto
-    OPENSSL_110 = {{ `command -v pkg-config > /dev/null && pkg-config --atleast-version=1.1.0 libcrypto || printf %s false`.stringify != "false" }}
-    OPENSSL_102 = {{ `command -v pkg-config > /dev/null && pkg-config --atleast-version=1.0.2 libcrypto || printf %s false`.stringify != "false" }}
+    # An extra zero is appended to the output of LIBRESSL_VERSION to make it 0 when LibreSSL does not exist on the system.
+    # Any comparisons to it should be affixed with an extra zero as well e.g. `(LIBRESSL_VERSION_NUMBER >= 0x2050500F0)`.
+    LIBRESSL_VERSION = {{ system("echo \"#include <openssl/opensslv.h>\nLIBRESSL_VERSION_NUMBER\" | " +
+                                 (env("CC") || "cc") + " " + `command -v pkg-config > /dev/null && pkg-config --cflags --silence-errors libssl || true`.chomp.stringify + " -E -").chomp.split('\n').last.split('L').first.id + "0" }}
+    OPENSSL_VERSION = {{ system("echo \"#include <openssl/opensslv.h>\nOPENSSL_VERSION_NUMBER\" | " +
+                                (env("CC") || "cc") + " " + `command -v pkg-config > /dev/null && pkg-config --cflags --silence-errors libssl || true`.chomp.stringify + " -E -").chomp.split('\n').last.split('L').first.id }}
```

Comment:
Why not `+ "0"`?

Reply:
Because `OPENSSL_VERSION_NUMBER` is assumed to be defined for *both* OpenSSL and LibreSSL. That's not the case for `LIBRESSL_VERSION_NUMBER` that isn't defined with OpenSSL.